handler/area_types.go

```diff
@@ -0,0 +1,52 @@
+package handler
+
+import (
+	"net/http"
+
+	"github.com/ONSdigital/dp-cantabular-dimension-api/contract"
+	"github.com/ONSdigital/dp-cantabular-dimension-api/model"
+	"github.com/pkg/errors"
+)
+
+const (
+	ParamCantabularBlob = "cantabular_blob"
+)
+
+// AreaTypes handles requests to /area-types
+type AreaTypes struct {
+	respond responder
+	ctblr   cantabularClient
+}
+
+// NewAreaTypes returns a new AreaTypes handler
+func NewAreaTypes(r responder, c cantabularClient) *AreaTypes {
+	return &AreaTypes{
+		respond: r,
+		ctblr:   c,
+	}
+}
+
+// Get is the handler for GET /area-types
+func (at *AreaTypes) Get(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	blob := r.URL.Query().Get(ParamCantabularBlob)
+
+	res, err := at.ctblr.GetGeographyDimensions(ctx, blob)
+	if err != nil {
+		at.respond.Error(ctx, w, errors.Wrap(err, "failed to get area-types from cantabular"))
```

Comment:
don't you need to log the error?

Reply:
The error gets logged by the responder

Reply:
it's automatically logged by a library:
Example, if you run the service with `make debug` and try a failing request:
```
{
  "created_at": "2022-02-04T13:08:26.504652Z",
  "event": "http request received",
  "http": {
    "method": "GET",
    "path": "/area-types",
    "query": "cantabular_blob=Example",
    "started_at": "2022-02-04T13:08:26.504646Z",
    "status_code": 0
  },
  "namespace": "dp-cantabular-dimension-api",
  "severity": 3
}
{
  "created_at": "2022-02-04T13:08:26.792535Z",
  "event": "http request completed",
  "http": {
    "duration": 287886000,
    "ended_at": "2022-02-04T13:08:26.792532Z",
    "method": "GET",
    "path": "/area-types",
    "query": "cantabular_blob=Example",
    "response_content_length": 41,
    "started_at": "2022-02-04T13:08:26.504646Z",
    "status_code": 500
  },
  "namespace": "dp-cantabular-dimension-api",
  "severity": 3
}

```

Reply:
well, you are right, it doesnt' log the error itself, only the request.  Iwill add a log for the error.

Reply:
I hate the "log at every point in the call stack" pattern but it seems to be the convention here